osm2streets/src/transform/collapse_short_road.rs

```diff
@@ -15,36 +12,32 @@ use crate::{
 
 impl StreetNetwork {
     /// Collapses a road, merging the two intersections together. Returns (the surviving
-    /// intersection, the deleted intersection, deleted roads, new roads)
+    /// intersection, the deleted intersection). The caller can see all roads connected to the
+    /// surviving intersection.
```

Comment:
Another massive change here, please check my logic carefully. These two "transforms" are some of the more complicated logic in the library so far

Reply:
The logic modifying the intersections and roads seems sound, but I haven't attempted to grok the trim distance or turn restriction logic.